server/auth-middleware.js

```diff
@@ -0,0 +1,21 @@
+require('dotenv/config');
+const { JsonWebTokenError, ...jwt } = require('jsonwebtoken');
+
+module.exports = function authMiddleware(req, res, next) {
+  const token = req.header('x-access-token');
+  if (!token) {
+    req.userId = null;
+    next();
+    return;
+  }
+
+  try {
+    // * Payload: {userId, username}
+    const { userId } = jwt.verify(token, process.env.TOKEN_SECRET);
+    req.userId = userId;
+  } catch (error) {
+    throw new JsonWebTokenError();
+  }
```

Comment:
You can actually strip out this try/catch here because if `verify` fails, it's going to throw a `JsonWebTokenError` anyway.

Reply:
oh sweet ok